avocado/core/plugin_interfaces.py

```diff
@@ -213,7 +213,14 @@ class VarianterPlugin(Plugin):
     @abc.abstractmethod
     def __iter__(self):
         """
-        Yield variant in form of (variant_id, (variant, mux_path))
+        Yields all variants
```

Comment:
Sure, as I mentioned earlier the later commits were added because the original code was not reviewed before I finished this commit. I'm fine with squashing those...